src/main/java/de/deepshore/kafka/Xsd2avroController.java

```diff
@@ -105,12 +75,63 @@ public HttpResponse convert(@Body XsdPack xsdpack) {
             return HttpResponse.ok(ap);
 
         } catch (NullPointerException npe) {
-            LOG.info("Error while converting XSD to AVRO", npe);
-            return HttpResponse.ok(String.format("Error while converting XSD to AVRO: %s", npe.getLocalizedMessage()));
+            return logAndReturnServerError(npe);
+        } catch (Exception e) {
+            return logAndReturnServerError(e);
+        }
+
+    }
+
+    @Post(uri = "/connect/java")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public HttpResponse getJava(@Body @Valid XsdPack xsdpack) {
+
+        try(FromXml.Value transform = new FromXml.Value()) {
+
+            final Map<String, String> transformConfig = createTransformConfig(xsdpack, this.getClass().getPackageName());
+
+            transform.configure(transformConfig);
+
+            File tempFile = File.createTempFile("compressed-java", String.valueOf(LocalDateTime.now()));
+
+            try(final FileOutputStream fos = new FileOutputStream(tempFile);
+                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
+                List<File> generatedSourceFiles = transform.getGeneratedSourceFiles();
+
+                generatedSourceFiles.forEach(srcFile -> {
+                    try(FileInputStream fis = new FileInputStream(srcFile)) {
+                        final ZipEntry zipEntry = new ZipEntry(srcFile.getName());
+                        zipOut.putNextEntry(zipEntry);
+                        byte[] bytes = new byte[1024];
+                        int length;
+                        while((length = fis.read(bytes)) >= 0) {
+                            zipOut.write(bytes, 0, length);
+                        }
+                    } catch (IOException e) {
+                        LOG.warn("Error zipping", e);
+                        return;
+                    }
+                });
+
+
+                return HttpResponse.ok(new StreamedFile(new FileInputStream(tempFile), new MediaType(MediaType.APPLICATION_OCTET_STREAM)))
+                        .header("Content-type", MediaType.APPLICATION_OCTET_STREAM)
+                        .header("Content-disposition", "attachment; filename=\"java.zip\"");
+            }
+        } catch (IOException ioe) {
+            LOG.info("Error while converting java files to zip", ioe);
+            return HttpResponse.serverError(String.format("Error while converting java files to zip: %s", ioe.getLocalizedMessage()));
+        } catch (NullPointerException npe) {
+            return logAndReturnServerError(npe);
         } catch (Exception e) {
```

Comment:
Unterscheidung zwischen NPE und Exception wird an diese Stelle nicht mehr gebraucht.

```suggestion
        } catch (Exception e) {
         // We want to catch every kind of Exception, since the client waits for a server response
```